parsl/usage_tracking/usage.py

```diff
@@ -98,90 +99,121 @@ def __init__(self, dfk, port=50077,
         self.domain_name = domain_name
         # The sock timeout will only apply to UDP send and not domain resolution
         self.sock_timeout = 5
-        self.UDP_PORT = port
+        self.udp_port = port
         self.procs = []
         self.dfk = dfk
         self.config = self.dfk.config
-        self.uuid = str(uuid.uuid4())
+        self.correlator_uuid = str(uuid.uuid4())
         self.parsl_version = PARSL_VERSION
-        self.python_version = "{}.{}.{}".format(sys.version_info.major,
-                                                sys.version_info.minor,
-                                                sys.version_info.micro)
-        self.tracking_enabled = self.check_tracking_enabled()
-        logger.debug("Tracking status: {}".format(self.tracking_enabled))
+        self.python_version = (
+            f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
+        )
+        self.tracking_level = self.check_tracking_level()
+        self.start_time = None
+        logger.debug(f"Tracking level: {self.tracking_level}")
+
+    def check_tracking_level(self):
+        """Check if tracking is enabled and return level.
+
+        Returns: int
+            - 0 : Tracking is disabled
+            - 1 : Tracking is enabled with level 1
+                  Share info about Parsl version, Python version, platform
+            - 2 : Tracking is enabled with level 2
+                  Share info about config + level 1
+            - 3 : Tracking is enabled with level 3
+                  Share info about app count, app fails, execution time + level 2
+        """
```

Comment:
can this code ever return a different value than the above?

Reply:
No.
config will raise an Exception if usage_tracking values are other than 0 (or False), 1 (or True), 2, 3
envvar can only be 'true', '1', '2' or '3'. Anything else is treated as level 0.